api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/staticapi/StaticDefinitionGenerator.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.apicatalog.staticapi;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.zowe.apiml.apicatalog.discovery.DiscoveryConfigProperties;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+
+/**
+ * Service to handle the creation of the static definition file
+ * Allows the generation and the override of the .yml. Retrieves the static definition location from the System env
+ * variables and stores the file there.
+ */
+@Service
+@Slf4j
+public class StaticDefinitionGenerator extends StaticAPIService {
+
+    private AtomicReference<String> fileName = new AtomicReference<>("");
+
+    @Value("${apiml.discovery.userid:eureka}")
+    private String eurekaUserid;
+
+    @Value("${apiml.discovery.password:password}")
+    private String eurekaPassword;
+
+    @Value("${server.attls.enabled:false}")
+    private boolean isAttlsEnabled;
+
+    public StaticDefinitionGenerator(RestTemplate restTemplate, DiscoveryConfigProperties discoveryConfigProperties) {
+        super(restTemplate, discoveryConfigProperties);
+    }
+
+
+    public StaticAPIResponse generateFile(String file) throws IOException {
+        String serviceId = StringUtils.substringBetween(file, "serviceId: ", "\\n");
```

Comment:
This is not the best way to get value out of json. It assumes formatted json which might not be the case. Better use something like JsonPath.